src/parsers/trades/gate.io/index.ts

```diff
@@ -0,0 +1,93 @@
+import { getCSVData } from '../../';
+import { EXCHANGES, IImport, IPartialTrade, ITrade } from '../../../types';
+import { createDateAsUTC, createID } from '../../utils';
+
+interface IGateIO {
+    No: string;
+    'Account Type': string;
+    Time: string;
+    'Action type': string;
+    Currency: string;
+    'Order id': string;
+    'Change amount': string;
+    Amount: string;
+    'Additional Info': string;
+}
+
+export default async function processData(importDetails: IImport): Promise<ITrade[]> {
+    const data: IGateIO[] = await getCSVData(importDetails.data) as IGateIO[];
+    const internalFormat: ITrade[] = [];
+    if (data.length < 1) {
+        return internalFormat;
+    }
+    let feeTrade = data[0];
```

Comment:
Why are we setting all of these to 0?

Reply:
I was getting filled trade is read before assigned at line 53 `tradeToAdd.boughtCurrency = filledTrade.Currency;` so I assign it to something first. It should be assigned in the switch structure but that does not compile.

Reply:
Okay so just to confirm Im understanding this right basically the reason your doing this is because 2 lines = 1 trade. One line is for the bought amount, then another for the sold amount right?

Reply:
Trades are on three lines and "Points Purchase" are on two lines, yes

Reply:
One of the lines of trades is the fee and the fee currency.
I had configured to pay the fee in POINT, which is not the bought currency nor the sold currency.
I thought that transactionFee was for that but you told me it’s not. So how to reflect this trade with 3 currencies?

Reply:
So I mentioned on another PR about using a for loop and just doing 2 or 3 lines at once. What are your thoughts on that?

And re POINTS, so actually that isnt supported in the current design at all so we would have to tweak things to make that supported. Maybe we can just add additional tradeFee and tradeCurrency field and default it to 0 for exist trades to keep backwards compatibility

Reply:
Yes, an alternative is to use a for loop with a counter for line.

That seems right to me to add tradeFee and tradeCurrency to the trade definition. That is quite a change though we would have to check in all the calculations whether it has to be deduced from holdings or not.

A temporary workaround would be to declare the trade fee as a trade between that fee and 0 or the same currency or 0 of USDT or something.

Reply:
Yeah that would be a workaround but Id really prefer to avoid that as then it makes linking those trade fees to the original trade

Reply:
Can’t we use transactionFee and transactionFeeCurrency also for tradeFee and tradeFeeCurrency?
The ideas are quite similar. To detect if it is a transactionFee or a tradeFee, we can test if one of the currencies are fiat.

Reply:
They are similiar but serve different purposes. For instance transactionFee is more intended for on chain exchanges. So like trading USDC to some other token and paying for the transaction with ETH

Reply:
Okay then let's implement tradeFee. See a strart in https://github.com/starsoccer/cryptotithe/pull/281

Reply:
I am going to make a discussion on it first as I dont want to implement this until some of the wasm stuff Ive done is merged in as that will cause breaking changing

Reply:
I pushed a better management of trades over multiples lines. And a basic support for tradeFee, at least for import